scaaml/capture/scope/cw_scope.py

```diff
@@ -43,29 +46,23 @@ def __init__(self, gain: int, samples: int, offset: int, clock: int,
               'sample_rate': sample_rate,
               'other_information': 'Can also be present.',
           }
-          with SScope(**capture_info) as scope:
+          with CWScope(**capture_info) as scope:
               # Use the scope object.
         """
-        self._scope = None
+        super().__init__(samples=samples, offset=offset)
         self._gain = gain
-        self._samples = samples
-        self._offset = offset
         self._clock = clock
         self._sample_rate = sample_rate
         self._basic_mode = "rising_edge"
         self._triggers = "tio4"
         self._freq_ctr_src = "clkgen"
         self._presamples = 0
+        self._scope = None
 
     def __enter__(self):
         """Create scope context.
 
-        Args:
-          gain: Gain of the scope.
-          samples: How many points to sample (length of the capture).
-          offset: Number of samples to wait before starting recording data.
-          clock: CLKGEN output frequency (in Hz).
-          sample_rate: Clock source for cw.ClockSettings.adc_src.
+        Returns: self
         """
         assert self._scope is None  # Do not allow nested with.
         cwscope = cw.scope()
```

Comment:
I think here you should directly assign `self._scope`.
Because I'm afraid that if an exception occurs while updating the settings, the context manager may call `__exit__()` and `self._scope` would still be `None`.